### DISCUSSION TITLE
Disable animation using view system?

### DISCUSSION THREAD

**Discussion by bthaler1** [patrykandpatrick/vico#556] [Questions] [answered]
Is it possible to disable the animation when adding a new entry in the view system? I'm not using compose. Or would it be possible to animate my line chart from the previous y to the new entry's y, rather then animate from 0?

**Comment by Gowsky** [comment, marked as answer]
Hi @bthaler1, you can disable difference animations by setting their duration to `0`. In Vico 1 you can do so by calling `ChartView#setDiffAnimationDuration(0L)` (or `ComposedChartView#setDiffAnimationDuration(0L)`). In Vico 2 you can do so by calling `CartesianChartView#setDiffAnimationDuration(0L)`.
Regarding animating an updated y-axis value for a given x-axis value, that's the default behavior you can see in the sample app. Please make sure that you're not recreating your `ChartEntryModelProducer`, `ComposedChartEntryModelProducer` (in Vico 1), or `CartesianChartModelProducer` (in Vico 2) in between data updates.
If it's still not working for you, please share your code with us and we'll try to help you.

**Comment by bthaler1** [comment]
Thanks @Gowsky.
setDiffAnimationduration(0L) worked perfectly for me.

Regarding the second question, let me rephrase. I don't want to update a y-axis value at an existing x-axis value. I'm adding a new x-axis value.

For example, say currently my max x-axis value is 10 (furthest right on the chart), and the corresponding y-axis value is 100. I then add another entry with an x-axis value of 11 and a y-axis value of 70. The chart animates the addition of the new entry from 0 to 70. What I would like to be able to do is animate from 100 to 70, if that makes sense.

I am not recreating the ChartEntryModelProducer. Also I am using Vico 1. Thanks for the help!

**Reply by Gowsky** [reply]
Okay, sorry for my misunderstanding. Yes, it's possible, but your `LineChart` requires a custom [`DrawingModelInterpolator`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.entry.diff/-drawing-model-interpolator/) implementation. The custom implementation may be based on the [`DefaultDrawingModelInterpolator`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.entry.diff/-default-drawing-model-interpolator/?query=class%20DefaultDrawingModelInterpolator%3CT%20:%20DrawingModel.DrawingInfo,%20R%20:%20DrawingModel%3CT%3E%3E%20:%20DrawingModelInterpolator%3CT,%20R%3E), so I'd suggest copying its code and modifying it. In the `updateTransformationMap()` ([here](https://github.com/patrykandpatrick/vico/blob/bda03006ea22574058983f155ec594ee633b61f8/vico/core/src/main/java/com/patrykandpatrick/vico/core/entry/diff/DefaultDrawingModelInterpolator.kt#L66)), for the last loop iteration (newly added entry) it should be enough to call `TransformationModel(map[x - 1]?.old, drawingInfo)` (use `x-1` if your x-axis step is `1`) to use old (previous) model's last x-axis value position as a starting point for the difference animation. Let me know if it helps.

**Comment by bthaler1** [comment]
Perfect, thanks. I will work on implementing this!
